### ISSUE TITLE
rfc5322 + exim dkim error (LONG_LINE)

### ISSUE THREAD

**Issue by alexeyantropov** [marlam/msmtp-mirror#53]
Exim (as relay for msmtp) fails DKIM with error "LONG_LINE" if it gets "special" e-main with wrong format. Wrong format is  looong lines in looong e-main, e.g. 300 lines with 5000 symbols. It may be some log from cronjob.

```
2021-05-24 13:39:07.234 [25791] 1ll7zH-0006hz-7W DKIM: validation error: LONG_LINE
2021-05-24 13:39:07.550 [25791] 1ll7zH-0006hz-7W DKIM: Error during validation, disabling signature verification: LONG_LINE
```

For example, Postfix uses smtp_line_length_limit config option to add line breaks. You can see it [here on Github](https://github.com/vdukhovni/postfix/blob/2595917e491dfe704390b9bf1100bcdd35b21ae8/postfix/src/smtp/smtp_proto.c#L1168).

How to simulate it.

- Config for msmtp /etc/msmtprc: 

```
defaults
tls_trust_file /etc/pki/tls/certs/ca-bundle.crt

account demo-relay
host SOME-EXIM-SERVER-AS-RELAY.domain.tld
port 25
from example@domain.tld

account default : demo-relay
```

- Prepare the test message
```
cat <<EOF >> /tmp/messsage.txt
Content-Type: text/plain; charset=utf-8
Subject: There is a bug
From: test@domain.tld

Hello!
EOF

for j in {1..300}; do for i in {1..5000}; do echo -n $j$i; done; echo; done >>  /tmp/messsage.txt
```

- send it with msmtp like with sendmail
```
cat  /tmp/messsage.txt | msmtp alexeyantropov@my.domain.tld
```

You can see same error in exim log after it. Also you can read more about this problem here: [exim-users at mail-archive.com](https://www.mail-archive.com/search?l=exim-users@exim.org&q=subject:%22%5C%5Bexim%5C%5D+DKIM%22&o=newest&f=1). I suggest to use same method for line breaks to avoid this error in remote relay.

**Comment by marlam** [comment]
Since the problem occurs in Exim and not in msmtp, I would suggest to change Exim instead.

Also, one of the principles of msmtp has always been to not alter the mail in any way (because whenever a mail program thinks it knows better, it will inevitably break someones use case). A few exceptions regarding headers had to be made, but we never changed a mail body so far.

**Comment by alexeyantropov** [comment]
Yes, I understand it. Different MTAs use different way for it. Some MTAs reject such e-mails with 550 Error. I attach [link](https://datatracker.ietf.org/doc/html/rfc5322#section-2.1.1) to RFC for you, if you are interesting.

**Comment by marlam** [comment]
Msmtp does not have a line length limit. The line length limit is in Exim, which is the receiving side here, and the RFC says "Receiving implementations would do well to handle an arbitrarily large number of characters in a line for robustness sake."
I think that's enough grounds to file a bug report with Exim; I don't think it makes sense for msmtp to try and work around limitations of Exim.
